common/lib/xmodule/xmodule/modulestore/tests/test_split_mongo_call_count.py

```diff
@@ -0,0 +1,119 @@
+                                target_course_id=source_course_key,
+                                create_course_if_not_present=True,
+                                raise_on_failure=True,
+                            )
+
+                        with check_mongo_calls(37):
+                            export_to_xml(
+                                source_store,
+                                source_content,
+                                source_course_key,
+                                self.export_dir,
+                                'exported_source_course',
+                            )
+
+                        with check_mongo_calls(16, 189):
+                            import_from_xml(
+                                dest_store,
+                                'test_user',
+                                self.export_dir,
+                                course_dirs=['exported_source_course'],
+                                static_content_store=dest_content,
+                                target_course_id=dest_course_key,
+                                create_course_if_not_present=True,
+                                raise_on_failure=True,
+                            )
+
+
+class CountMongoCallsCourseTraversal(TestCase):
+    """
+    Tests the number of Mongo calls made when traversing a course tree from the top course root
+    to the leaf nodes.
+    """
+
+    def test_number_mongo_calls(self):
+        # Construct the contentstore for storing the course import
+        with MongoContentstoreBuilder().build() as source_content:
+            # Construct the modulestore for storing the course import (using the previously created contentstore)
+            with MIXED_SPLIT_MODULESTORE_BUILDER.build(source_content) as source_store:
+
+                source_course_key = source_store.make_course_key('a', 'course', 'course')
+
+                # First, import a course.
+                import_from_xml(
+                    source_store,
+                    'test_user',
+                    TEST_DATA_DIR,
+                    course_dirs=['manual-testing-complete'],
+                    static_content_store=source_content,
+                    target_course_id=source_course_key,
+                    create_course_if_not_present=True,
+                    raise_on_failure=True,
+                )
+
+                # Course traversal modeled after the traversal done here:
+                # lms/djangoapps/mobile_api/video_outlines/serializers.py:BlockOutline
+                # Starting at the root course block, do a breadth-first traversal using
+                # get_children() to retrieve each block's children.
+                # pylint: disable=bad-continuation
+                for depth, num_calls in (
+                    (None, 7),  # The way this traversal *should* be done.
+                    (0, 145)    # The pathological case - do *not* query a course this way!
+                ):
```

Comment:
The other advantage to `ddt` is that it gives you two different test cases (ideal and pathological) that can fail independently. With the current setup, it's less obvious when a test fails which version of the test is broken.